docs/reference/multisite.rst

```diff
@@ -100,38 +80,16 @@ The last action is to configure the ``sonata_page`` section as:
 Host and Path By Locale Strategy
 --------------------------------
 
-This strategy requires a dedicated ``RequestFactory`` object. So you need to alter the
-front controller to use the one provided by the ``PageBundle``.
-
-To do so, open ``index.php`` file and change the following parts::
-
-    // public/index.php
-
-    use App\Kernel;
-    use Symfony\Component\ErrorHandler\Debug;
-    use Sonata\PageBundle\Request\RequestFactory; # before: use Symfony\Component\HttpFoundation\Request;
-
-    require dirname(__DIR__).'/config/bootstrap.php';
-
-    if ($_SERVER['APP_DEBUG']) {
-        umask(0000);
-
-        Debug::enable();
-    }
+This strategy requires a dedicated runtime. So you need to modify your
+composer.json and declare ``SonataPageRuntime`` runtime.
 
-    if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? $_ENV['TRUSTED_PROXIES'] ?? false) {
-        Request::setTrustedProxies(explode(',', $trustedProxies), Request::HEADER_X_FORWARDED_ALL ^ Request::HEADER_X_FORWARDED_HOST);
-    }
+.. code-block:: json
 
-    if ($trustedHosts = $_SERVER['TRUSTED_HOSTS'] ?? $_ENV['TRUSTED_HOSTS'] ?? false) {
-        Request::setTrustedHosts([$trustedHosts]);
-    }
+        "extra": {
+            "runtime": {
+                "class": "Sonata\\PageBundle\\Runtime\\SonataPageRuntime"
+            }
 
-    $kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
-    $request = RequestFactory::createFromGlobals('host_with_path_by_locale'); // before: $request = Request::createFromGlobals();
```

Comment:
Does `RequestFactory::createFromGlobals` still useful or can it be deprecated @eerison ?

Reply:
The whole RequestFactory class can be deprecated.